kano_peripherals/wrappers/detection.py

```diff
@@ -112,7 +146,42 @@ def is_ck2_pro(with_dbus=True, retry_count=5):
     Returns:
         bool: Whether this is a CK2 Pro or not
     """
-    return is_power_hat_plugged(with_dbus=with_dbus, retry_count=retry_count)
+    hat_connected = is_power_hat_plugged(
+        with_dbus=with_dbus, retry_count=retry_count
+    )
+
+    if not hat_connected:
+        return False
+
+    kit, dummy_version = get_screen_version()
+
+    return kit == CKC
+
+
+def is_ckt(with_dbus=True, retry_count=5):
+    """Check if the hardware is recognised as a Computer Kit Touch.
+
+
+    The determination is based on the PowerHat board being plugged in along with
+    a matching EDID for the CKT kits.
+
+    Args:
+        with_dbus: See :func:`is_power_hat_plugged`
+        retry_count: See :func:`is_power_hat_plugged`
+
+    Returns:
+        bool: Whether this is a CKT or not
+    """
+    hat_connected = is_power_hat_plugged(
```

Comment:
Can this be simplified? if !hat_connected && kit == CKT, then surely it is still a CKT, so maybe this step is redundant.

Reply:
Hmm, you're right, it probably is. It is only because our previous detection method was to just check if the hat was connected, nothing to do with EDID, and now we need to distinguish between these cases. If we think that we are happy going by EDID alone then I'm happy to remove this check.